apps/user-office-backend/src/models/questionTypes/InstrumentPicker.ts

```diff
@@ -0,0 +1,143 @@
+      if (field.question.dataType !== DataType.INSTRUMENT_PICKER) {
+        throw new GraphQLError('DataType should be INSTRUMENT_PICKER');
+      }
+
+      return new Promise((resolve) => resolve(true));
+    },
+    createBlankConfig: (): InstrumentPickerConfig => {
+      const config = new InstrumentPickerConfig();
+      config.small_label = '';
+      config.required = false;
+      config.tooltip = '';
+      config.variant = 'dropdown';
+      config.instruments = [];
+
+      return config;
+    },
+    getDefaultAnswer: () => null,
+    filterQuery: (queryBuilder, filter) => {
+      const value = JSON.parse(filter.value).value;
+      switch (filter.compareOperator) {
+        case QuestionFilterCompareOperator.EQUALS:
+          return queryBuilder.andWhereRaw("answers.answer->>'value'=?", value);
+        default:
+          throw new GraphQLError(
+            `Unsupported comparator for SelectionFromOptions ${filter.compareOperator}`
+          );
+      }
+    },
+    transformConfig: async (config, callId) => {
+      const fallBackConfig = { ...config, instruments: [] };
+
+      try {
+        const instruments = await database
+          .select([
+            'i.instrument_id',
+            'name',
+            'short_code',
+            'description',
+            'manager_user_id',
+            'chi.availability_time',
+            'chi.submitted',
+          ])
+          .from('instruments as i')
+          .join('call_has_instruments as chi', {
+            'i.instrument_id': 'chi.instrument_id',
+          })
+          .where('chi.call_id', callId)
+          .distinct('i.instrument_id')
+          .then((instruments: InstrumentWithAvailabilityTimeRecord[]) => {
+            return instruments;
+          });
+
+        return {
+          ...config,
+          instruments: instruments.map(
+            (instrument) =>
+              new InstrumentOptionClass(
+                instrument.instrument_id,
+                instrument.name
+              )
+          ),
+        };
+      } catch (err) {
+        logger.logError('Call Instruments fetch failed', {
+          err,
+        });
+      }
+
+      return fallBackConfig;
+    },
+    afterSave: async (questionaryId, value) => {
+      // Get Proposal
+      const proposal = await database
+        .select()
+        .from('proposals')
+        .where('questionary_id', questionaryId)
+        .first()
+        .then((proposal: ProposalRecord | null) =>
+          proposal ? createProposalObject(proposal) : null
+        );
+      // Get Instrument
+      const instrument = await database
+        .select()
+        .from('instruments')
+        .where('instrument_id', value)
+        .first()
+        .then((instrument: InstrumentRecord | null) =>
+          instrument ? createInstrumentObject(instrument) : null
+        );
+      if (!instrument || !proposal) return;
+      await database.transaction(async (trx) => {
```

Comment:
I am not sure about doing the change this way. Because like this it is changed in the database but no event is fired. If we have let's say a trigger on the event PROPOSAL_INSTRUMENT_SELECTED somewhere I think it won't be triggered at all in this case. So instead of directly interacting with the database this should go trough the mutations where the events are set.